rca/api_content/content.py

```diff
@@ -53,11 +55,14 @@ def try_and_fetch(url):
         error_text = "Error occured when fetching further detail data"
         logger.exception(error_text)
         raise CantPullFromRcaApi(error_text)
+    else:
+        return None
```

Comment:
When used in a `try` statement, `else` runs if an exception _isn't_ raised, so this line won't run. See https://geek-university.com/python/the-try-except-else-statements/

Its best to have as few statements in a `try` as possible, just the ones that might raise, then have the safer ones in the `else`.

So (assuming the function doesn't re-raise):

```
def fetch_data(url):
    logger.info(f"pulling data from API {url}")
    try:
        response = requests.get(url, timeout=5)
        response.raise_for_status()
    except Timeout:
        logger.exception(f"Timeout error occurred when fetching data from {url}")
    except (HTTPError, ConnectionError):
        logger.exception(f"Connection error occurred when fetching further detail data from {url}")
    else:
        return response.json()
```

Reply:
Thanks, I've neatened this up a bit in https://github.com/torchbox/rca-wagtail-2019/pull/313/commits/bb89344e2606ee15594bdbad3ac506c5f7b4aae4

Reply:
I've added an exception for anything that is not specified in https://github.com/torchbox/rca-wagtail-2019/pull/313/commits/7b7f88507f886ead17917bb1faa5626ef1b8a16f

I've tested raising theses locally and making sure the page loads with no data should any of the exceptions get caught by raising them in the code at the `try` point. All loads ok

Reply:
The answer to this was that the custom exception `CantPullFromRca` is caught here https://github.com/torchbox/rca-wagtail-2019/blob/master/rca/home/models.py#L186